controllers/addresspool_controller.go

```diff
@@ -47,48 +50,105 @@ const (
 // +kubebuilder:rbac:groups=metallb.io,resources=addresspools/status,verbs=get;update;patch
 
 func (r *AddressPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("addresspool", req.NamespacedName)
-	log.Info("Reconciling AddressPool resource")
+	r.Log.Info(fmt.Sprintf("Starting AddressPool reconcile loop for %v", req.NamespacedName))
 
 	instance := &metallbv1alpha1.AddressPool{}
+	defer r.Log.Info(fmt.Sprintf("Finish AddressPool reconcile loop for %v", req.NamespacedName))
+
 	if err := r.Get(ctx, req.NamespacedName, instance); err != nil {
+		if errors.IsNotFound(err) {
+			err = r.syncMetallbAddressPools(req)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	err := r.syncMetalLBAddressPool(instance)
 	if err != nil {
-		errors.Wrap(err, "Failed to create address-pool config map")
+		r.Log.Info(fmt.Sprintf("sync MetalLB addresspool failed %s", err))
 		return ctrl.Result{RequeueAfter: RetryPeriod}, err
 	}
 
-	log.Info("Reconcile complete")
 	return ctrl.Result{}, nil
 }
 
-func (r *AddressPoolReconciler) syncMetalLBAddressPool(instance *metallbv1alpha1.AddressPool) error {
+func renderObject(instance *metallbv1alpha1.AddressPool) ([]*unstructured.Unstructured, error) {
 	data := render.MakeRenderData()
 	data.Data["Name"] = instance.Spec.Name
 	data.Data["Protocol"] = instance.Spec.Protocol
 	data.Data["AutoAssign"] = *instance.Spec.AutoAssign
 	data.Data["Addresses"] = instance.Spec.Addresses
 	objs, err := render.RenderDir(AddressPoolManifestPath, &data)
 	if err != nil {
-		return errors.Wrapf(err, "Fail to render address-pool manifest")
+		return nil, fmt.Errorf("Fail to render address-pool manifest %v", err)
```

Comment:
In this case, I believe it's safe to assume that objs will contain one single object (the configmap), so I't better to check for the len and return the first element (which makes the name a bit less confusing)